packages/core/src/serialization/SerializationContext.ts

```diff
@@ -78,6 +78,9 @@ export class SerializationContext<T> {
   isMarkedAsPopulated(entityName: string, prop: string): boolean {
     let populate: PopulateOptions<T>[] | undefined = this.populate;
 
+    if (populate.length === 1 && populate[0].field === '*') {
```

Comment:
maybe this would be better this way? i think there are valid use cases for `*` on different positions too

```suggestion
    if (populate.some(({ field }) === '*') {
```